commons/src/main/java/org/infinispan/commons/util/ReflectionUtil.java

```diff
@@ -128,6 +133,21 @@ private static void inspectRecursively(Class<?> c, List<Method> s, Class<? exten
       }
    }
 
+   private static void inspectFieldsRecursively(Class<?> c, List<Field> s, Class<? extends Annotation> annotationType) {
+      if (c == null || c.isInterface()) {
+         return;
+      }
+      for (Field f : c.getDeclaredFields()) {
+         if (f.isAnnotationPresent(annotationType)) {
+            s.add(f);
+         }
+      }
+
+      if (!c.equals(Object.class)) {
```

Comment:
It's safe to use ```==``` here.

Reply:
True, should I change it in `inspectRecursively` above, too?